table/column.go

```diff
@@ -360,27 +360,36 @@ func getColDefaultValue(ctx sessionctx.Context, col *model.ColumnInfo, defaultVa
 		}
 		return value, nil
 	}
+
 	// Check and get timestamp/datetime default value.
-	value, err := expression.GetTimeValue(ctx, defaultVal, col.Tp, col.Decimal)
-	if err != nil {
-		return types.Datum{}, errGetDefaultFailed.GenWithStack("Field '%s' get default value fail - %s",
-			col.Name, errors.Trace(err))
-	}
-	// If column is timestamp, and default value is not current_timestamp, should convert the default value to the current session time zone.
+	sc := ctx.GetSessionVars().StmtCtx
+	var needChangeTimeZone bool
+	// If the column's default value is not ZeroDatetimeStr nor CurrentTimestamp, should use the time zone of the default value itself.
 	if col.Tp == mysql.TypeTimestamp {
 		if vv, ok := defaultVal.(string); ok && vv != types.ZeroDatetimeStr && strings.ToUpper(vv) != strings.ToUpper(ast.CurrentTimestamp) {
-			t := value.GetMysqlTime()
+			needChangeTimeZone = true
+			originalTZ := sc.TimeZone
 			// For col.Version = 0, the timezone information of default value is already lost, so use the system timezone as the default value timezone.
-			defaultTimeZone := timeutil.SystemLocation()
+			sc.TimeZone = timeutil.SystemLocation()
 			if col.Version >= model.ColumnInfoVersion1 {
-				defaultTimeZone = time.UTC
-			}
-			err = t.ConvertTimeZone(defaultTimeZone, ctx.GetSessionVars().Location())
-			if err != nil {
-				return value, errors.Trace(err)
+				sc.TimeZone = time.UTC
```

Comment:
Why do we need to change sc.TimeZone?

Reply:
Because when `col.Version >= ColumnInfoVersion1`(#9115) is true, the default value uses the UTC as the timezone.

Reply:
Please add some comments here, explain sc.TimeZone is used by expression.GetTimeValue, so this the reason why we change the sc.TimeZone.

Reply:
Maybe we should add a new parameter to pass the timezone to GetTimeValue? I affaid the defer will infect the performance.

Reply:
It seems that getTimeValue(ctx, xxx) will use sc.

Reply:
But the check function deep nested in `GetTimeValueis`, and these functions used a lot of places.  Besides, this change is a bit strange, there is already timezone in `sc`, we add a time zone parameter